src/com/goide/editor/GoStatementMover.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2013-2016 Sergey Ignatov, Alexander Zolotov, Florin Patan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.editor;
+
+import com.goide.psi.GoFile;
+import com.goide.psi.GoImportDeclaration;
+import com.goide.psi.GoImportList;
+import com.goide.psi.GoTopLevelDeclaration;
+import com.intellij.codeInsight.editorActions.moveUpDown.LineMover;
+import com.intellij.codeInsight.editorActions.moveUpDown.LineRange;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.util.Pair;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.util.PsiTreeUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+
+public class GoStatementMover extends LineMover {
+  @Override
+  public boolean checkAvailable(@NotNull Editor editor, @NotNull PsiFile file, @NotNull MoveInfo info, boolean down) {
+    if (!(file instanceof GoFile || super.checkAvailable(editor, file, info, down))) return false;
+
+    Pair<PsiElement, PsiElement> primeElementRange = getElementRange(editor, file, getLineRangeFromSelection(editor));
+    if (primeElementRange == null) return false;
+    PsiElement commonParent =
+      primeElementRange.first.isEquivalentTo(primeElementRange.second) ? primeElementRange.first.getParent() :
+      PsiTreeUtil.findCommonParent(primeElementRange.first, primeElementRange.second);
+    if (commonParent == null) return false;
+
+    Pair<PsiElement, PsiElement> elementRange = getLogicElementRange(primeElementRange, commonParent);
+    if (elementRange == null) return false;
+    if (commonParent == elementRange.first) commonParent = commonParent.getParent();
+    info.toMove = new LineRange(elementRange.first, elementRange.second);
+    if (elementRange.first instanceof GoTopLevelDeclaration) {
+      if (!file.isValid()) {
+        return true;
```

Comment:
Why do you checking the file?
Why do you tell platform that your line mover is an appropriate for invalid file?